tests/Agent/IntegrationTests/SharedApplications/Common/MultiFunctionApplicationHelpers/NetStandardLibraries/MsSql/MicrosoftDataSqlClientExerciser.cs

```diff
@@ -0,0 +1,293 @@
+                var deleteSql = string.Format(DeletePersonMsSql, tableName);
+
+                using (var command = new SqlCommand(insertSql, connection))
+                {
+                    var insertCount = command.ExecuteNonQuery();
+                }
+
+                using (var command = new SqlCommand(countSql, connection))
+                {
+                    var teamMemberCount = command.ExecuteScalar();
+                }
+
+                using (var command = new SqlCommand(deleteSql, connection))
+                {
+                    var deleteCount = command.ExecuteNonQuery();
+                }
+            }
+
+            return string.Join(",", teamMembers);
+        }
+
+        [LibraryMethod]
+        [Transaction]
+        [MethodImpl(MethodImplOptions.NoOptimization | MethodImplOptions.NoInlining)]
+        public async Task<string> MsSqlAsync(string tableName)
+        {
+            var teamMembers = new List<string>();
+
+            using (var connection = new SqlConnection(MsSqlConfiguration.MsSqlConnectionString))
+            {
+                connection.Open();
+
+                using (var command = new SqlCommand("SELECT * FROM NewRelic.dbo.TeamMembers WHERE FirstName = 'John'", connection))
+                {
+                    using (var reader = await command.ExecuteReaderAsync())
+                    {
+                        while (await reader.ReadAsync())
+                        {
+                            teamMembers.Add(reader.GetString(reader.GetOrdinal("FirstName")));
+                            if (await reader.NextResultAsync())
+                            {
+                                teamMembers.Add(reader.GetString(reader.GetOrdinal("FirstName")));
+                            }
+                        }
+                    }
+                }
+
+                var insertSql = string.Format(InsertPersonMsSql, tableName);
+                var countSql = string.Format(CountPersonMsSql, tableName);
+                var deleteSql = string.Format(DeletePersonMsSql, tableName);
+
+                using (var command = new SqlCommand(insertSql, connection))
+                {
+                    var insertCount = await command.ExecuteNonQueryAsync();
+                }
+
+                using (var command = new SqlCommand(countSql, connection))
+                {
+                    var teamMemberCount = await command.ExecuteScalarAsync();
+                }
+
+                using (var command = new SqlCommand(deleteSql, connection))
+                {
+                    var deleteCount = await command.ExecuteNonQueryAsync();
+                }
+            }
+
+            return string.Join(",", teamMembers);
+        }
+
+        [LibraryMethod]
+        [Transaction]
+        [MethodImpl(MethodImplOptions.NoOptimization | MethodImplOptions.NoInlining)]
+        public string MsSql_WithParameterizedQuery(bool paramsWithAtSign)
```

Comment:
Minor nitpick: I don't think we should use underscores in the names of the library methods